Check/Codebase/PhpCodeSniffer.php

```diff
@@ -0,0 +1,171 @@
+/**
+ * Class SiteAuditCheckCodebasePhpCodeSniffer.
+ */
+class SiteAuditCheckCodebasePhpCodeSniffer extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('PHP Code Sniffer Violations');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Run phpcs on custom code.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {
+    if (isset($this->registry['phpcs_path'])) {
+      return dt('Cannot find phpcs in path. Make sure that phpcs is present in the PATH or run composer install inside the site_audit directory to install all the dependencies');
+    }
+    if (isset($this->registry['custom_code'])) {
+      return dt('No custom code path specified');
+    }
+    return dt('Cannot find coding standards inside ' . $this->registry['phpcs_standard']);
+  }
+
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('No PHP code_sniffer violations.');
+  }
+
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {
+    $ret_val = '';
+    if (drush_get_option('html') == TRUE) {
+      $ret_val .= '<table class="table table-condensed">';
+      $ret_val .= '<thead><tr><th>' . dt('Line, Column') . '</th><th>' . dt('Severity') . '</th><th>' . dt('Action') . '</th></tr></thead>';
+      foreach ($this->registry['phpcs_out'] as $filename => $violations) {
+        $num_violations = count($violations);
+        $ret_val .= "<tr align='center'><td colspan='3'><a href='file:///$filename'>File: $filename</a> Violations: $num_violations</td></tr>";
+        foreach ($violations as $violation) {
+          $line = $violation['line'];
+          $column = $violation['column'];
+          $severity = $violation['severity'];
+          $message = $violation['message'];
+          $ret_val .= "<tr><td>$line, $column</td><td>$severity</td><td>$message</td></tr>";
+        }
+      }
+      $ret_val .= '</table>';
+    }
+    else {
+      $rows = 0;
+      foreach ($this->registry['phpcs_out'] as $filename => $violations) {
+        if ($rows++ > 0) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 4);
+          }
+        }
+        $ret_val .= dt('Filename: @filename, Violations: @total', array(
+          '@filename' => $filename,
+          '@total' => count($violations),
+        ));
+        foreach ($violations as $violation) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 6);
+          }
+          $line = $violation['line'];
```

Comment:
Is this really necessary? Keep It Simple, there are no bonus points for extra lines :-)